src/query/functions/src/scalars/map.rs

```diff
@@ -244,4 +252,144 @@ pub fn register(registry: &mut FunctionRegistry) {
                     .any(|(k, _)| k == key)
             },
         );
+
+    registry.register_3_arg_core::<EmptyMapType, GenericType<0>, GenericType<1>, MapType<GenericType<0>, GenericType<1>>, _, _>(
+        "map_insert",
+        |_, _, insert_key_domain, insert_value_domain| {
+            FunctionDomain::Domain(Some((
+                insert_key_domain.clone(),
+                insert_value_domain.clone(),
+            )))
+        },
+        |_, key, value, ctx| {
```

Comment:
use `vectorize_with_builder_3_arg` here, so that we can support column values in this function